### PR TITLE
[RTM] Remove module checks

### PR DESCRIPTION
**Pull request by aschempp**
These checks should be removed. They are not compatible with the current ResourceProvider and they won't be compatible with the new solution I'm working on.

I just extracted that so we could *discuss* that separately.

### CODE REVIEW

**Comment by leofeyer** [comment]
We can remove neither of them. It will still be possible to install the ER2 as a legacy bundle, so we have to keep the first check. And as long as we support using `database.sql` files, we have to keep the second check, too.

**Comment by aschempp** [comment]
I disagree. Installing the repository should not be supported. Obviously people could try to extract the extension from Contao 3, but that would result in templates showing up. Fine with me…

Regarding the second, that's a hack because the same problem would affect every extension but it's hardcoded to the core ones. Also, we agreed that it should be possible to run Contao 3 extensions in Contao 4, but we don't want to put efforts in an update process. This "fix" would only affect systems updated from Contao 2.

File: src/Resources/contao/dca/tl_templates.php [hunk 1]
```diff
@@ -227,7 +227,7 @@ public function addNewTemplate()
 		foreach ($kernel->getContainer()->get('contao.resource_provider')->getResourcesPaths() as $strModule => $strFolder)
 		{
 			// Continue if there is no templates folder
-			if ($strModule == 'repository' || !is_dir($strFolder))
```
Comment (leofeyer) [inline src/Resources/contao/dca/tl_templates.php]: I think this is wrong. Isn't `$strFolder` the Contao resource path? Should it not be `$strFolder . '/templates`?

Comment (aschempp) [inline src/Resources/contao/dca/tl_templates.php]: You're right, but this is already fixed in my PR to `resource_locator`.

**Comment by leofeyer** [comment]
> Also, we agreed that it should be possible to run Contao 3 extensions in Contao 4, but we don't want to put efforts in an update process.

That's not true. We don't want to put efforts in an update process right now, but certainly soon.

> This "fix" would only affect systems updated from Contao 2.

That's not true either. The `database.sql` files are supported in Contao 3, too. They have not been deprecated yet.

File: src/Resources/contao/dca/tl_templates.php (hunk 1 above)
Comment (leofeyer) [inline src/Resources/contao/dca/tl_templates.php]: Can I still ask you to fix it here? It's kind of confusing to merge faulty code.

Comment (aschempp) [inline src/Resources/contao/dca/tl_templates.php]: That will result in a conflict in the other branch…

**Comment by aschempp** [comment]
> That's not true either. The database.sql files are supported in Contao 3, too. They have not been deprecated yet.

Correct, but that's only for the core modules if they contain SQL files. And that can only be the case for a Contao 2 upgrade.

**Pull request closed without merging**